src/node-jspm-jasmine.js

```diff
@@ -141,17 +180,73 @@ export function runTests(opts, errCallback = function() {}) {
 							importTheseTestFiles();
 						}
 					})
-					.catch(errCallback);
+					.catch((ex) => {
+						if (opts.coverage && opts.coverage.tempDirectory) {
+							// remove temporary directory
+							rimraf.sync(opts.coverage.tempDirectory)
+						}
+						errCallback(ex)
+					});
 				});
 			});
 		}
 	} catch(ex) {
-		removeTempFiles();
+		if (opts.coverage && opts.coverage.tempDirectory) {
+			// remove temporary directory
+			rimraf.sync(opts.coverage.tempDirectory)
+		}
 		errCallback(new Error(`Jasmine or Jspm may not be properly configured -- '${ex.toString()}'`));
 	}
 }
 
 function importTestFiles(SystemJS, jasmine, specDir, specFiles, coverage, errCallback) {
+
+	if (coverage) {
+		// the "onComplete" hook to prevent jasmine's self-righteous exit
+		jasmine.onComplete(function (passed) {
+			// avoid misalignment with jasmine's output
+			console.log('');
+			// import the rest of the modules not already imported (evaluated)
+			// as dependencies of specs
+			Promise.all(Object.keys(coverage.coverageFiles).map(function (file) {
+				return SystemJS.import(path.join(process.cwd(), file));
+			})).then(function () {
+				if (typeof __coverage__ === 'undefined') {
+					console.log("No coverage was collected for files matching globs " + coverage.files);
+				} else {
+					try {
+						const collector = remap(__coverage__)
+						let report = Report.create(
+							(coverage.reporter || 'html'),
+							{
+								dir: coverage.dir
+							}
+						);
+						report.writeReport(collector, true);
```

Comment:
It is not present in the master branch, just in this pr. But I don't think that it is actually a problem in your code here -- I think that is actually related to trying to do coverage on all the files instead of just some of the files. I'll try it out now and see if it works. Thanks for all your work on this and your patience with me on combing through this pr over the last few days.